apps/api/views/files.py

```diff
@@ -19,15 +18,16 @@ def render(self, data, accepted_media_type=None, renderer_context=None):
 
 
 @extend_schema(operation_id="file_content", summary="Download File Content", tags=["Files"], responses=bytes)
-@api_view(["GET"])
-@renderer_classes([BinaryRenderer])
-@permission_required("files.view_file")
-def file_content_view(request, pk: int):
-    file = get_object_or_404(File, id=pk, team=request.team)
-    if not file.file:
-        raise Http404()
-
-    try:
-        return FileResponse(file.file.open(), as_attachment=True, filename=file.file.name)
-    except FileNotFoundError:
-        raise Http404() from None
+class FileContentView(APIView):
+    required_scopes = ("sessions:read", "chatbots:read")
```

Comment:
According to the docs, the `TokenHasScope` class will require the token to have ALL the scopes listed: https://django-oauth-toolkit.readthedocs.io/en/latest/rest-framework/permissions.html#tokenhasscope

Is that the intention?

Reply:
+1 for new scope